lib/bolt_server/transport_app.rb

```diff
@@ -703,6 +703,9 @@ def make_winrm_target(target_hash)
           connect_plugin = BoltServer::Plugin::PuppetConnectData.new(body['puppet_connect_data'])
           plugins = Bolt::Plugin.setup(context[:config], context[:pal], load_plugins: false)
           plugins.add_plugin(connect_plugin)
+          %w[aws_inventory azure_inventory gcloud_inventory].each do |plugin_name|
+            plugins.add_module_plugin(plugin_name) if plugins.modules.include?(plugin_name)
```

Comment:
Oh, what happens if the user overrides one of these modules, the override is no longer a module plugin, and they're not using that module as an inventory plugin [meaning the override makes sense]? E.g. I have an `aws_inventory` module that is not a module plugin and I am not using the `aws_inventory` plugin in my inventory file. I think in this case, `add_module_plugin` would error whereas in Bolt, I think we'd just pass-thru b/c dynamic loading would ensure that the plugin's never loaded and hence `add_module_plugin(aws_inventory)` is never called.

If that is a valid case, do we want to handle it? If so, how?

Do we want to handle that here

Reply:
I think that means this check should change from `plugins.modules.include?(plugin_name)` to `plugins.modules.include?(plugin_name) && plugins.modules[plugin_name].plugin?`. Or just `plugins.known_plugin?(plugin_name)` because we _know_ these aren't Ruby plugins.

Reply:
Updated!